File: apps/android/src/main/java/com/rustella/stellartrail/domain/gear/GearOptions.kt
package com.rustella.stellartrail.domain.gear

import com.rustella.stellartrail.domain.atlas.GearAtlasStatus

val GearCategory.label: String
    get() = when (this) {
        GearCategory.BACKPACK_SYSTEM -> "背负系统"
        GearCategory.SLEEP_SYSTEM -> "睡眠系统"
        GearCategory.KITCHEN_SYSTEM -> "餐厨系统"
        GearCategory.WALKING_SYSTEM -> "行走系统"
        GearCategory.CLOTHING_SYSTEM -> "衣物系统"
        GearCategory.LIGHTING_SYSTEM -> "照明系统"
        GearCategory.FIRST_AID_SYSTEM -> "急救系统"
        GearCategory.ELECTRONICS_SYSTEM -> "电子系统"
        GearCategory.TECHNICAL_GEAR -> "技术装备"
        GearCategory.OTHER_GEAR -> "其它装备"
        GearCategory.CONSUMABLE -> "消耗品"
    }

val GearStatus.label: String
    get() = when (this) {
        GearStatus.AVAILABLE -> "可用"
        GearStatus.IN_USE -> "使用中"
        GearStatus.MAINTENANCE -> "保养中"
        GearStatus.DAMAGED -> "损坏"
        GearStatus.LOST -> "遗失"
        GearStatus.RETIRED -> "退役"
        GearStatus.SOLD -> "已售出"
        GearStatus.IDLE -> "闲置"
    }

val GearShareStatus.label: String
    get() = when (this) {
        GearShareStatus.NOT_SHARED -> "未共享"
        GearShareStatus.PENDING -> "待审核"
        GearShareStatus.APPROVED -> "已通过"
        GearShareStatus.REJECTED -> "已拒绝"
        GearShareStatus.WITHDRAWN -> "已撤回"
    }

val GearAtlasStatus.label: String
    get() = when (this) {
        GearAtlasStatus.PENDING -> "审核中"
        GearAtlasStatus.APPROVED -> "已收录"
        GearAtlasStatus.REJECTED -> "已拒绝"
    }

val GearSort.label: String
    get() = when (this) {
        GearSort.CREATED_AT_DESC -> "添加时间由新到旧"
        GearSort.CREATED_AT_ASC -> "添加时间由旧到新"
        GearSort.PURCHASE_DATE_DESC -> "购买日期由新到旧"
        GearSort.NAME_ASC -> "装备名称 A-Z"
        GearSort.WEIGHT_DESC -> "重量由高到低"
        GearSort.PRICE_DESC -> "价格由高到低"
    }

val GearTab.label: String
    get() = when (this) {
        GearTab.AVAILABLE -> "可用装备"
        GearTab.HISTORY -> "历史装备"
    }

val allGearCategories: List<GearCategory> = GearCategory.entries
val allGearStatuses: List<GearStatus> = GearStatus.entries
val allGearSorts: List<GearSort> = GearSort.entries

File: apps/web/src/gear-options.ts
import type {
  GearCategory,
  GearSort,
  GearStatus,
} from "@stellartrail/shared-types";

export const CATEGORY_OPTIONS: Array<{ value: GearCategory; label: string }> = [
  { value: "backpack_system", label: "背负系统" },
  { value: "sleep_system", label: "睡眠系统" },
  { value: "kitchen_system", label: "餐厨系统" },
  { value: "walking_system", label: "行走系统" },
  { value: "clothing_system", label: "衣物系统" },
  { value: "lighting_system", label: "照明系统" },
  { value: "first_aid_system", label: "急救系统" },
  { value: "electronics_system", label: "电子系统" },
  { value: "technical_gear", label: "技术装备" },
  { value: "other_gear", label: "其它装备" },
  { value: "consumable", label: "消耗品" },
];

export const STATUS_OPTIONS: Array<{ value: GearStatus; label: string }> = [
  { value: "available", label: "可用" },
  { value: "in_use", label: "使用中" },
  { value: "maintenance", label: "保养中" },
  { value: "damaged", label: "损坏" },
  { value: "lost", label: "遗失" },
  { value: "retired", label: "退役" },
  { value: "sold", label: "已售出" },
  { value: "idle", label: "闲置" },
];

export const SORT_OPTIONS: Array<{ value: GearSort; label: string }> = [
  { value: "created_at_desc", label: "添加时间由新到旧" },
  { value: "created_at_asc", label: "添加时间由旧到新" },
  { value: "purchase_date_desc", label: "购买日期由新到旧" },
  { value: "name_asc", label: "装备名称 A-Z" },
  { value: "weight_desc", label: "重量由高到低" },
  { value: "price_desc", label: "价格由高到低" },
];

export function categoryLabel(value: GearCategory): string {
  return CATEGORY_OPTIONS.find((item) => item.value === value)?.label ?? value;
}

export function statusLabel(value: GearStatus): string {
  return STATUS_OPTIONS.find((item) => item.value === value)?.label ?? value;
}
